fhir-bulkdata-webapp/src/main/java/com/ibm/fhir/bulkdata/jbatch/export/checkpoint/ExportCheckpointAlgorithm.java

```diff
@@ -39,6 +53,18 @@ public int checkpointTimeout() {
 
     @Override
     public void beginCheckpoint() {
+        if (isFileExport == null) {
+            JobOperator jobOperator = BatchRuntime.getJobOperator();
+            JobExecution jobExecution = jobOperator.getJobExecution(jobCtx.getExecutionId());
+            BatchContextAdapter contextAdapter = new BatchContextAdapter(jobExecution.getJobParameters());
+            BulkDataContext ctx = contextAdapter.getStepContextForSystemChunkWriter();
+            isFileExport = "file".equals(ConfigurationFactory.getInstance().getStorageProviderType(ctx.getSource()));
+        }
+
+        if (adapter == null) {
+            adapter = ConfigurationFactory.getInstance();
+        }
```

Comment:
Two thoughts... put this ahead of line 56, and then call adapter in isFileExport. 

You can also inline it to line 43, it can be setup when the class is initialized as it is stateless

Reply:
moved to line 43 as suggested